### PR TITLE
Update deps to reduce the vulnerabilities, migrate to Gulp 4

### PR DESCRIPTION
**Pull request by TotomInc**
## Brief description

#153, PaperCSS have more than 100+ vulnerabilities because of outdated (dev-)dependencies when running `npm audit`.

There are now 6 vulnerabilities:

- 5 criticals (**Regular Expression Denial of Service**): caused by `minimatch` in `gulp-watch-less`/`gulp-watch`, in `simplebuild-karma`and `simplebuild-jshint`.

- 1 low (**Prototype Pollution**): caused by `lodash` in `gulp-watch-less`/`gulp-watch`.

The project have also migrated to Gulp 4 which have done a big re-design in their API.

## Developer Certificate of Origin

- [x] I certify that these changes according to the Developer Certificate of Origin 1.1 as described at <https://developercertificate.org/>.

### Further details

I have removed the `yarn.lock` since it is **not** recommended to have multiple lock files and use multiple package managers. We will stay with npm as it is the default and official package manager for NodeJS.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -21,51 +21,49 @@ gulp.task('sass', function() {
...
-      ]
+        { formatter: 'string', console: true },
+      ],
     }));
 })
 
-gulp.task('default', ['sass','watch','hugo-server']);
-gulp.task('build', ['sass','minify-css','hugo-build']);
-gulp.task('postinstall', ['sass','minify-css']);
+gulp.task('default', gulp.series('sass', 'minify-css', 'hugo-build'));
```
Comment (rhyneav) [inline gulpfile.js]: Ah nice, so `gulp.series` is the replacement for passing in an array?

Comment (TotomInc) [reply gulpfile.js]: Absolutelty, there are `gulp.series` and `gulp.parallel`. I've followed [this simple guide](https://fettblog.eu/gulp-4-parallel-and-series/) for the migration.

**Comment by TotomInc** [comment]
After further investigation, the remaining vulnerabilities can't really be fixed except by the dependencies themselves.

This includes:

- `gulp-watch-less`: 3 high vulnerabilities caused by a sub-module called `minimatch`.
- `simplebuild-jshint`: 1 high vulnerability caused by a sub-module called `minimatch`.
- `simplebuild-karma`: 1 high vulnerability caused by a sub-module called `minimatch`.

File: gulpfile.js (hunk 1 above)
Comment (rhyneav) [reply gulpfile.js]: Awesome!

**Pull request merged**